message.go

```diff
@@ -22,6 +22,9 @@ type Message struct {
 	// writing the message.
 	Time time.Time
 
+	// Message attributes.
+	Attributes int8
```

Comment:
Why do we need this? Isn't the `CompressionCodec` enough?

Reply:
We need it when we decompress the message we've just consumed. `CompressionCodec` is populated when we create a message to be produced.

Reply:
Decoding is done here right? https://github.com/segmentio/kafka-go/blob/c087815a5658518dd07e2bab939de0db8921adea/batch.go#L173

Seems like we could pass the attributes to `decode` instead of exposing an implementation detail in the public API of `Message`.

Reply:
We are also calling the `decode` method in the `Reader` where we don't have access to the attributes (I think, I will double check).